app/services/claims/claim_window/build.rb

```diff
@@ -0,0 +1,30 @@
+class Claims::ClaimWindow::Build
+  include ServicePattern
+
+  def initialize(claim_window_params:, claim_window: nil)
+    @claim_window = claim_window
+    @claim_window_params = claim_window_params
+  end
+
+  def call
+    self.claim_window ||= Claims::ClaimWindow.new
+    self.claim_window.assign_attributes(claim_window_params)
+    self.claim_window.academic_year = find_academic_year
+    self.claim_window
+  end
+
+  private
+
+  attr_accessor :claim_window, :claim_window_params
+
+  def find_academic_year
+    return unless claim_window.starts_on
```

Comment:
Not sure this is necessary. As ClaimWindow validates that a `starts_on` is present.

Reply:
This is necessary, as the `Build#call` does not validate the `ClaimWindow` within it's method scope.

Alternatively, we _could_ do:

```ruby
claim_window = ...
claim_window.academic_year = find_academic_year if claim_window.valid?
```

but that's pulling in external dependencies to the `find_academic_year`. The academic year only depends on a `claim_window.starts_on` date being present.